### ISSUE TITLE
OWL files seem empty so can't convert to OBO

### ISSUE THREAD

**Issue by ChristopherMancuso** [obophenotype/upheno#534]
We are going to start doing some work in our group comparing transcriptome samples across a number of species. It looks like the upheno ontology is exactly what we are looking for! The 6 species we are going to look at are mouse, human, fly, worm, zebrafish and yeast, which according to the GitHub page (https://github.com/obophenotype/upheno#contributing-ontologies) are the ontologies that contribute to upheno. However, I am having some trouble finding out how to download the upheno.owl and convert it to an obo. Basically whenever I download the file, it looks like the file just has a header but no relationship information about the nodes in the ontology. I have tried downloading the upheno.owl from:
 
Github https://github.com/obophenotype/upheno/blob/master/upheno.owl
OBO foundry http://purl.obolibrary.org/obo/upheno.owl
OntoBee http://purl.obolibrary.org/obo/upheno.owl (same as OBO foundry)
AberOWL http://aber-owl.net/media/ontologies/UPHENO/3/upheno.owl
 
I am using robot to convert from owl to obo. The robot program seems to work as I can take the GO and PhenomeNet owl files and convert them to obo files. I download those two files at: 
 
GO http://purl.obolibrary.org/obo/go.owl
PhenonmeNet http://aber-owl.net/media/ontologies/PhenomeNET/1/phenomenet.owl
 
When I download the upheno owl I get something like
 
```
  <?xml version="1.0"?>
  
  
 <!DOCTYPE rdf:RDF [
     <!ENTITY owl "http://www.w3.org/2002/07/owl#" >
     <!ENTITY obo "http://purl.obolibrary.org/obo/" >
     <!ENTITY xsd "http://www.w3.org/2001/XMLSchema#" >
     <!ENTITY rdfs "http://www.w3.org/2000/01/rdf-schema#" >
     <!ENTITY rdf "http://www.w3.org/1999/02/22-rdf-syntax-ns#" >
     <!ENTITY oboInOwl "http://www.geneontology.org/formats/oboInOwl#" >
 ]>
 
 
 <rdf:RDF xmlns="&obo;x-bfo.owl#"
      xml:base="&obo;x-bfo.owl"
      xmlns:obo="http://purl.obolibrary.org/obo/"
      xmlns:rdfs="http://www.w3.org/2000/01/rdf-schema#"
      xmlns:owl="http://www.w3.org/2002/07/owl#"
      xmlns:xsd="http://www.w3.org/2001/XMLSchema#"
      xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
      xmlns:oboInOwl="http://www.geneontology.org/formats/oboInOwl#">
     <owl:Ontology rdf:about="&obo;upheno.owl">
         <owl:imports rdf:resource="&obo;upheno/vertebrate.owl"/>
     </owl:Ontology>
 </rdf:RDF> 


 
 <!-- Generated by the OWL API (version 3.4.2) http://owlapi.sourceforge.net -->
``` 

If you could let me know how to get the owl or obo files that contain information from mouse, human, fly, worm, zebrafish and yeast that would be great. Thanks in advance for any help!

**Comment by matentzn** [comment]
@ChristopherMancuso I will respond to you soon; thanks for the patience. :)

**Comment by balhoff** [comment]
@ChristopherMancuso the upheno ontology imports the others. ROBOT is downloading these when it loads the file, but when you save to OBO it will keep these as imported rather than merged together. You can use `robot merge` to combine before saving to OBO. In the case of the upheno ontologies, I suspect you will be missing a fair amount of information by converting to OBO (this may not matter for your application). The command is:

```
robot merge -I 'http://purl.obolibrary.org/obo/upheno.owl' -o upheno.obo
```

I would recommend always using the ontology PURL so that you get the main release (for any OBO library ontology). This is the obolibrary URI: `http://purl.obolibrary.org/obo/upheno.owl`

For GO you can download an OBO file directly, from: `http://purl.obolibrary.org/obo/go.obo`

There are a few issues to mention:
- The upheno file seems to only import the vertebrate ontologies at the moment. @matentzn can advise on the best approach to get the full set.
- When you run the ROBOT command I pasted above, you will get an error: `OBO STRUCTURE ERROR Ontology does not conform to OBO structure rules`. This is because the merged ontology violates some constraints (e.g. multiple definitions for a single term). You can instead save to the [obographs JSON format](https://douroucouli.wordpress.com/2016/10/04/a-developer-friendly-json-exchange-format-for-ontologies/), if there is any way you can work with that format: `robot merge -I 'http://purl.obolibrary.org/obo/upheno.owl' -o upheno.json` (or use an OWL parser)
- If you try to use an OWL reasoner on the merged ontologies, you will encounter many problems. If you want to do this, let us know, and we can point you to some versions of these files that work better for this use case (the main release files may contain conflicting versions of some shared content).

**Comment by matentzn** [comment]
Hahha @balhoff thanks for writing most of what I wanted to say!

@ChristopherMancuso So the first thing is: upheno as you can find it on this repository is a bit outdated, and will be replaced by a [completely new version](https://github.com/obophenotype/upheno-dev/) in the first quarter of 2020. This version has much cleaner hierarchies; if you open the current upheno there are a lot of dangling classes that are not integrated into class hierarchy, which makes it a bit harder to use. In any case: The Monarch Initiative uses the file called metazoa.owl in the upheno repository rather than upheno.owl; it contains some more ontologies like c elegans phenotypes and some bridges. I have quickly generated obo (heavily stripped down because of the srtcture rule violations, just to take a look at) and json versions for you to look at [here](https://www.dropbox.com/s/sjt3gdpv89o820r/upheno.zip?dl=0). But you can use the approach that @balhoff suggested to do it yourself.

In any case, my suggestion is to play a bit with the ontology and see how far you get; I am happy to help! Stay tuned for upheno2 which will come with more species coverage and much cleaner hierarchies.

**Comment by ChristopherMancuso** [comment]
Thanks @balhoff and @matentzn! This has definitely got me headed in the right direction. I was able to generate the files using robot and I will look at those and the files @matentzn sent in the link and see how far I cant get. I for sure need to read more about owl ontologies in general. I think it is a little different than what I was thinking would happen. For instance I was thinking if you merged the phenotype ontologies for say human and zebrafish you would end up with a completely new but self consistent node IDs where if a node in the human ontology was considered the same as a  node in the zebrafish ontology it would get the same node ID in the merged/new ontology. This is probably way too naive of a view, I've just starting getting into ontologies :)

Thanks again for the quick and awesome help with this!

**Comment by matentzn** [comment]
This is not naive at all, and, indeed, some ontologies like [Mondo](https://github.com/monarch-initiative/mondo) do exactly that: mint a new id for each disease and associate the reference ontologies using cross-references. The uPheno version you see here does not have any species independent phenotypes at all (or only very few); it is mainly a merge of the existing phenotype ontologies with a bit of bridging in between.  uPheno2 will have species independent phenotype classes like the ones you are thinking about; but rather than removing the species-specific classes altogether (like mondo), they will leave underneath their species independent counterparts, like so:

- abnormal limb morphology
   - abnormal limb morphology (fly)
   - abnormal limb morphology (mouse)

But you wont find this here!

**Comment by cmungall** [comment]
There is also an option for avoiding structure checks when saving obo

**Comment by matentzn** [comment]
Yeah thanks @cmungall i did exactly that for the files I attached above! :)

**Comment by balhoff** [comment]
Aha, I didn’t realize that was in robot. I was thinking only owltools.

**Comment by ChristopherMancuso** [comment]
Thanks for all the information, it has all been super helpful! I will be on the look out for upheno2 as well!
